backend/api/permissions.py

```diff
@@ -11,3 +11,21 @@ def has_permission(self, request, view):
             return True
         else:
             return request.user.is_staff
+
+
+class IsStaffOrCreateOnly(permissions.BasePermission):
+    """
+    Custom permission to only allow create and list actions.
+    """
+    def has_permission(self, request, view):
+        # Defining available request methods.
+        staff_methods = ("GET", "OPTIONS", "HEAD", "POST")
+        not_staff_methods = ("OPTIONS", "POST")
+
+        if request.user.is_staff and request.method in staff_methods:
+            # If the user is staff and the request method is among the allowed methods for staff.
```

Comment:
Is it just me, or is this kind of redundant? Personally I don't care for comments like these, but you can keep them if you want.